crates/script/src/runner.rs

```diff
@@ -148,6 +147,14 @@ impl ScriptRunner {
             self.executor.set_nonce(self.evm_opts.sender, u64::MAX / 2)?;
         }
 
+        // NOTE(zk): Address recomputed again after the nonce modification as it will
+        // differ in case of evm_opts.sender == CALLER
+        let zk_actual_script_address = CALLER.create(self.executor.get_nonce(CALLER)?);
```

Comment:
shouldn't we move the entire `set_balance` thign as well after this nonce modification? we also get to comput teh address only once
Maybe candidate for upstream patch

Reply:
Maybe but would prefer to take this to upstream see what they say, maybe try triggering a bug that tires to have an initial balance but fails due to this problem. Might also be there is some context we are missing.

Reply:
I mean, the commment specifies that it's setting the balance of the contract of the contract being deployed before the deployment itself, so I think it's desirable for the address to be the correct one instead of the previous calculation (which is incorrect)

Reply:
Yes but the upstream one has the same problem. Let's keep it as upstream and change it on upstream first if at all, again we may be missing context on why it is like this.